### PR TITLE
[DPE-4564][BUGFIX] Don't query nodes lock for nodes out of the relation (issue when removing unts)

### PR DESCRIPTION
**Pull request by juditnovak**
## Issue

https://github.com/canonical/opensearch-operator/issues/323

## Solution

The bugfix makes sure that nodes that may have left the relation (typically when removing units from the cluster) are not being searched for in the databag.

NOTE: I'm unsure how to test the functionality with the automated testsuite... I suggest that we could schedule https://github.com/canonical/opensearch-operator/issues/410 where a test could tackle the fix provided here?

### CODE REVIEW

**Review by reneradoi — DISMISSED** [review]

**Review by Mehdi-Bendriss — APPROVED** [review]
Thanks Judit!

File: tests/unit/lib/test_opensearch_lock.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            ]
+            == "true"
+        )
+
+        # A new event introduces a new Juju context ID
+        # At this point the leader can take the lock
+        os.environ["JUJU_CONTEXT_ID"] = "new-context-id"
+        assert self.harness.charm.node_lock.acquired
+
+    def test_node_lock_no_online_nodes_depating_node_doesnt_break(self):
```
Comment (reneradoi) [inline tests/unit/lib/test_opensearch_lock.py:138]: "departing"

File: tests/unit/lib/test_opensearch_lock.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        databag = self.harness.charm.model.get_relation(NodeLockRelationName).data
+        databag._data.pop(unit1)
+
+        # The property function executes healthy
+        # instead of breaking over the unit missing from the databag
+        assert not self.harness.charm.node_lock.acquired
+
+    @responses.activate
+    @patch("socket.socket.connect")
+    def test_node_lock_has_online_nodes_depating_node_doesnt_break(self, _):
```
Comment (reneradoi) [inline tests/unit/lib/test_opensearch_lock.py:164]: "departing"

**Review by reneradoi — APPROVED** [review]
Hey Judit, thank you for fixing the issue and also adding test coverage with this PR! All I've found is two typos, I've marked them with comments. Everything else is fine from my side.

File: tests/unit/lib/test_opensearch_lock.py (hunk 1 above)
Comment (juditnovak) [reply tests/unit/lib/test_opensearch_lock.py:138]: Thx much :-) 

Let me fix the typo in the "neighbor" PR (similar bugfix) (avoiding to loose 2 precious approvals ;-) )

File: tests/unit/lib/test_opensearch_lock.py (hunk 2 above)
Comment (juditnovak) [reply tests/unit/lib/test_opensearch_lock.py:164]: Let me fix the typo in the "neighbor" PR (similar bugfix) (avoiding to loose 2 precious approvals ;-) )

**Pull request merged**